Add files via upload

PavelGozhii opened the pull request.

PavelGozhii committed:
Add files via upload

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public static String[] getMostInputWords(String inputString) {
+        inputString = inputString.replaceAll("[,.]", "").toLowerCase();
```
there are other symbols which are not considered a word

Siril-O replied:
also pay attention words can be separated not only by spaces

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,50 @@
+        inputString = inputString.replaceAll("[,.]", "").toLowerCase();
+        String[] stringArr = inputString.split(" ");
+        List<Pair<String, Integer>> pairList = new ArrayList();
+        for (String currentString : stringArr) {
```
streams please

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,50 @@
+        return answer;
+    }
+
+    private static ArrayList removeDuplicates(ArrayList list) {
```
use streams, this can be done much more simpler

PavelGozhii replied:
> use streams, this can be done much more simpler

well I will think over the decision, just I do not see yet a trivial solution

PavelGozhii committed:
Update Main.java

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+
+    public static String[] newGetMostInputWordsN(String inputString) {
```
rename method

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,37 @@
+
+    public static String[] newGetMostInputWordsN(String inputString) {
+        List<String> answer = Stream.of(inputString.toLowerCase().split("(?:[!,.?^_]|-)"))
+                .flatMap(n -> Arrays.asList(n.split(" ")).stream())
```
just split by non word characters why do you need flatmap here ?

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,37 @@
+                .sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .collect(
+                        Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (n1, n2) -> {
+                                    throw new IllegalStateException();
```
Why do you need this exception here ?

Siril-O commented on Main.java:
```diff
@@ -0,0 +1,37 @@
+                .collect(Collectors.toList());
+
+        String[] strings = new String[10];
+        IntStream.range(0, answer.size()).forEach(i -> strings[i] = answer.get(i));
```
Why so difficult just print answer  in foreach ?

PavelGozhii committed:
Update Main.java

I completely redesigned it because the previous solution had too many  ̶c̶r̶u̶t̶c̶h̶e̶s̶ features
